### ISSUE TITLE
Readiness probes not working for onnx-tensorrt models

### ISSUE THREAD

**Issue by samueltrautwein** [triton-inference-server/server#8085]
**Description**
When running an onnx model on trition with tensorrt, the first requests are always timing out and inference requests are working as expected only 1-2 minutes after that.

Configuring model warmup did not resolve the problem. We're seeing in the prometheus metrics that those warmup requests are actually triggered.

**Triton Information**
We're deploying triton in kubernetes using the image `nvcr.io/nvidia/tritonserver:24.12-py3`.

**To Reproduce**

*Model* 

A fine-tuned fp16 variant of https://huggingface.co/microsoft/mdeberta-v3-base model converted to a `model.onnx`. No furher optimizations are applied

*Configuration*

```
name: "deberta"
platform: "onnxruntime_onnx"

max_batch_size: 16

dynamic_batching {
  preferred_batch_size: [ 1, 2, 4, 8, 16]
  max_queue_delay_microseconds: 0
}

instance_group [
  {
    count: 2
    kind: KIND_GPU
  }
]

optimization {
  # enable onnx graph optimizations
  graph : { level : 1 }

  # configure tensorrt execution provider for the onnx model
  execution_accelerators : {
    gpu_execution_accelerator : [{
      name : "tensorrt"
      parameters { key: "max_workspace_size_bytes" value: "8589934592" } # 8 GiB
      parameters { key: "trt_engine_cache_enable" value: "1" }
  }]
  }
}

# 0: sequential execution / 1: parallel
parameters { key: "execution_mode" value: { string_value: "0" } }


# model warmup configuration
model_warmup [{

  batch_size: 16
  count: 3

  inputs {
    key: "input_ids"
    value: {
      data_type: TYPE_INT64
      dims: [ 512 ]
      random_data: true
    }
  }

  inputs {
    key: "attention_mask"
    value {
      data_type: TYPE_INT64
      dims: [ 512 ]
      input_data_file: "attention_mask_ones.bin"
    }
  }
}]
```

*Integration / Clients*

Requests are generated done using the python http client (tritonclient  2.54.0) with async_infer.

**Expected behavior**

The `/v2/health/ready` endpoint should only return 200 when all models are responsive.

**PS:** I observed a similar bug where all model inference requests where failing with 5xx and the following error message: `TRTKernel_graph_main_graph_2725473138066827561_0 node. Name:'TensorrtExecutionProvider_TRTKernel_graph_main_graph_2725473138066827561_0_0' Status Message: No engine is found.` The problem in this situation was that all health checks kept returning a 200 response. 

I'm not sure if this issue is related to the one reported above - just adding it for more context.

**Comment by oneprofittips** [comment]
While tensor is being optimized for the onnx model, it returns that the model is available, but in fact, until the optimization passes, the request hangs in the queue indefinitely. The optimization takes about 10-25 minutes for the model.
